internal/juju/applications.go

```diff
@@ -173,21 +175,42 @@ func (input CreateApplicationInput) validateAndTransform() (parsed transformedCr
 	}
 	parsed.placement = placements
 
+	endpointBindings := map[string]string{}
```

Comment:
This functionality is not required when we use the DeployFromRepository API call, the endpoints will be validated by the controller. We're trying to have less business logic in the clients. 

Please move lines 178-195 to legacyDeploy(). Then `parsed.endpointBindings = input.endpointBindings` can be places at the beginning of the method.

Any juju controller version 3.3+ will use DeployFromRepository

Reply:
The reason I performed this check before hand is because I get a `panic: runtime error: invalid memory address or nil pointer dereference` when giving a space that does not exist.

I will open a bug on Juju

Reply:
https://bugs.launchpad.net/juju/+bug/2055868

Reply:
Please add a comment that we can remove the checks once the bug is fixed.